service.py

```
@@ -1,116 +1,420 @@
+            response = self.session.get(
+                "https://openapi.investec.com/za/pb/v1/accounts",
+                timeout=30
+            )
+            response.raise_for_status()
+            
+            self.accounts = response.json()["data"]["accounts"]
+            logger.info(f"Successfully fetched {len(self.accounts)} accounts")
+            
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to fetch accounts: {e}")
+            raise APIException(f"Failed to fetch accounts: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)